### ISSUE TITLE
`parcel [serve] --lazy` fails to load an async bundle under certain conditions

### ISSUE THREAD

**Issue by rayinaway** [parcel-bundler/parcel#8816]
# 🐛 bug report

`parcel [serve]` fails to load an async bundle under the following conditions:

- `--lazy` flag is set.
- Along with the async bundle, the parent module statically imports some other module.
- Parcel cache has not been created yet.

These conditions seem exhaustive, a minimal repro below reflects that. After the cache is created, the next HMR invocation resolves the issue.

## 🎛 Configuration

Apart from the `--lazy` flag, out-of-the-box Parcel.

## 🤔 Expected Behavior

When the async bundle is requested, Parcel's development server responds with it.

## 😯 Current Behavior

When the async bundle is requested, the server responds with _`index.html`_ as a fallback instead, and a `SyntaxError` from failed parsing is thrown.

```
dynamic.51313f62.js:1 Uncaught SyntaxError: Unexpected token '<'
index.63aff760.js:61 Uncaught (in promise) Error: Cannot find module 'kmWTC'
    at newRequire (index.63aff760.js:61:19)
    at Function.newRequire (index.63aff760.js:45:18)
    at runtime-647fb96160fc59b6.js:1:248
```

## 💁 Possible Solution

Might be related to how Parcel initializes the cache.

## 🔦 Context

Faced the issue when tried to dynamically import app-specific devtools under certain conditions only.

## 💻 Code Sample

https://github.com/rayinaway/lazy-serve-parcel-bug-repro

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.8.3
| Node             | 16.13.2
| npm         | 8.1.2
| Operating System | Windows 10

**Comment by marcins** [comment]
This should be fixed by #9093.

**Comment by marcins** [comment]
This should be fixed by https://github.com/parcel-bundler/parcel/pull/9093.
